front/lib/resources/storage/models/data_source_view.ts

```diff
@@ -75,7 +70,10 @@ DataSourceViewModel.init(
     indexes: [
       { fields: ["workspaceId", "id"] },
       { fields: ["workspaceId", "vaultId"] },
-      { fields: ["workspaceId", "dataSourceId", "vaultId"], unique: true },
+      {
```

Comment:
New index to allow data source views to remain accessible while the soft-deleted ones are being removed.

Reply:
Why do we need dataSourceId here ooc?

Reply:
We want to limit to one view per workspace, data source and vault.